twisted/scripts/_twistd_unix.py

```diff
@@ -198,6 +199,38 @@ def preApplication(self):
         self.oldstderr = sys.stderr
 
 
+    def _formatException(self, exception):
+        """
+        Format the C{exception} in preparation for writing to the
+        status pipe.  This does the right thing on Python 2 if the
+        exception's message is Unicode, and in all cases limits the
+        length of the message, prior to encoding, to 100 codepoints.
+
+        This means the return message may be longer than 100 bytes.
+
+        @type exception: L{Exception}
+        @param exception: The exception to format.
+
+        @return: The formatted message, suitable for writing to the
+            status pipe.
+        @rtype: L{bytes}
+        """
+        # On Python 2 this will encode Unicode messages with the ascii
+        # codec and the backslashreplace error handler.
+        exceptionLine = traceback.format_exception_only(exception.__class__,
+                                                        exception)[-1]
+        # remove exception name
+        exceptionMessage = exceptionLine.split(': ', 1)[-1]
```

Comment:
I'd prefer to leave the name in.  I didn't because I was worried about gratuitously changing the error message, since it's a kind of public API.  But I've already changed it a lot by using `_bytesRepr`, so I'm less worried about it now.  And I also think seeing what would be the last line of the traceback would be very helpful.

Should I ask on the mailing list if anybody has strong opinions about the format of this message?